### PR TITLE
fix(queue): get rid of repeat options from default

### LINKED ISSUE #1282: repeat option is ignored when using "defaultJobOptions"
Despite being allowed to use the "repeat" option when instantiating a Queue with "defaultJobOptions", the repeat options are ignored when adding the job. Since having "repeat" as default option is quite unusual I think the solution. for this bug is just to remove the repeat option from the allowed defaultJobOptions.

### PR DESCRIPTION
**Pull request by roggervalf**
fixes #1282

### CODE REVIEW

File: src/interfaces/jobs-options.ts [hunk 1]
```diff
@@ -94,6 +69,38 @@ export interface JobsOptions {
...
+  /**
+   * Base repeat job key.
+   */
+  repeatJobKey?: string;
+
+  /**
+   * The number of milliseconds after which the job should be
+   * fail with a timeout error.
+   */
+  timeout?: number;
```
Comment (roggervalf) [inline src/interfaces/jobs-options.ts:93]: timeout, I think is not used, could I delete it?

Comment (manast) [reply src/interfaces/jobs-options.ts:93]: Not used. But I thinkn it is used in the Pro version. So we need to add it to Pro then.

File: src/interfaces/jobs-options.ts [hunk 2]
```diff
@@ -94,6 +69,38 @@ export interface JobsOptions {
...
+export interface JobsOptions extends BaseJobOptions {
+  /**
+   * Repeat this job, for example based on a `cron` schedule.
+   */
+  repeat?: RepeatOptions;
+
+  /**
+   * Base repeat job key.
+   */
+  repeatJobKey?: string;
```
Comment (manast) [inline src/interfaces/jobs-options.ts:87]: Is this option really needed? I am thinking the user is not able to configure this key, it is computed automatically from the "repeat" option.

**Review by manast — COMMENTED** [review]
just one comment

File: src/interfaces/jobs-options.ts (hunk 2 above)
Comment (roggervalf) [reply src/interfaces/jobs-options.ts:87]: yes, in this case is like prevMillis option, I'll change the description as prevMillis saying that this is an internal param

**Review by manast — APPROVED** [review]

**Pull request merged**